erpnext/stock/utils.py

```diff
@@ -419,6 +419,17 @@ def is_reposting_item_valuation_in_progress():
 	if reposting_in_progress:
 		frappe.msgprint(_("Item valuation reposting in progress. Report might show incorrect item valuation."), alert=1)
 
+
+def calculate_mapped_packed_items_return(return_doc):
+	parent_items = set([item.parent_item for item in return_doc.packed_items])
+	against_doc = frappe.get_doc(return_doc.doctype, return_doc.return_against)
+	for d_item, item in zip(against_doc.items, return_doc.items):
+		if d_item.item_code in parent_items:
+			for p_item, d_p_item in zip(return_doc.packed_items, against_doc.packed_items):
```

Comment:
If you have two bundle this for loop adds last bundle to all references. It's not being checked which item they actually belong to here.

Reply:
original

<img width="1422" alt="Screenshot 2022-01-21 at 11 08 23 AM" src="https://user-images.githubusercontent.com/9079960/150472249-35c6d8c5-a447-4673-87fd-cbf6d834a5a6.png">



returned:
<img width="1422" alt="Screenshot 2022-01-21 at 11 08 44 AM" src="https://user-images.githubusercontent.com/9079960/150472257-ff0596b0-3457-4321-83b5-faa50e29bd41.png">